src/_pytest/pythonpath.py

```diff
@@ -6,14 +6,23 @@
 from pytest import Parser
 
 
+added_paths: List[str]
+
+
 def pytest_addoption(parser: Parser) -> None:
     parser.addini("pythonpath", type="paths", help="Add paths to sys.path", default=[])
 
 
 @pytest.hookimpl(tryfirst=True)
-def pytest_load_initial_conftests(
-    early_config: Config, parser: Parser, args: List[str]
-) -> None:
-    """`pythonpath = a b` will set `sys.path` to `[a, b, x, y, z, ...]`"""
-    for path in reversed(early_config.getini("pythonpath")):
-        sys.path.insert(0, str(path))
+def pytest_load_initial_conftests(early_config: Config) -> None:
+    global added_paths
+    added_paths = [str(p) for p in early_config.getini("pythonpath")]
+    # `pythonpath = a b` will set `sys.path` to `[a, b, x, y, z, ...]`
+    for path in reversed(added_paths):
+        sys.path.insert(0, path)
+
+
+def pytest_unconfigure() -> None:
+    global added_paths
```

Comment:
this creates issues with running nested pytest instances in pytester or running multiple pytest runs in the same process
i suggest using a lpugin

Reply:
In this case, we already have the information in config, so retrieving it again with `getini` seems simplest and symmetrical. I'll try it and see how it looks.

Reply:
I tried both methods. I've pushed the `getini` version.

For comparison, here's the `getini` version:
```python
@pytest.hookimpl(tryfirst=True)
def pytest_load_initial_conftests(early_config: Config) -> None:
    # `pythonpath = a b` will set `sys.path` to `[a, b, x, y, z, ...]`
    for path in reversed(early_config.getini("pythonpath")):
        sys.path.insert(0, str(path))


@pytest.hookimpl(trylast=True)
def pytest_unconfigure(config: Config) -> None:
    for path in config.getini("pythonpath"):
        sys.path.remove(path)
```

Here's a `config.stash` version.
```python
@pytest.hookimpl(tryfirst=True)
def pytest_load_initial_conftests(early_config: Config) -> None:
    early_config.stash[key] = [str(p) for p in early_config.getini("pythonpath")]
    # `pythonpath = a b` will set `sys.path` to `[a, b, x, y, z, ...]`
    for path in reversed(early_config.stash[key]):
        sys.path.insert(0, path)


@pytest.hookimpl(trylast=True)
def pytest_unconfigure(config: Config) -> None:
    if key in config.stash:
        for path in config.stash[key]:
            sys.path.remove(path)
        del config.stash[key]
```

I find the `getini` version more readable. 
However, both work.
Please let me know if there's a preference.

Reply:
Thanks @RonnyPfannschmidt for noticing the problem. I didn't like using a global variable.
And thanks @bluetech for the suggestions.